src/openforms/utils/json_logic.py

```diff
@@ -49,3 +52,174 @@ def normalize_value(value: Union[dict, list, tuple, int, str]) -> JSONLogicValue
+
+    :param logics:  :class:`JsonLogicTest` the logic rule
+    :param keys_mapped_to_component: dict of component with component key as key
+    :param keys_mapped_to_form: dict of form with component key as key
+    :param resulting_data: actual data from the submission form
+
+    Example of data structure returned:
+
+    {
+        "nodeType":"operation",
+        "operator":">",
+        "children":[
+            {
+                "nodeType":"leaf",
+                "key":"birthdate",
+                "actualValue":"2022-06-21",
+                "stepName":"definition-birthdate",
+                "label":"Birthdate"
+            }
+        ],
+        "value":"2022-06-20"
+    }
+
+    :rtype: dict
+    :return: rule with additional informations related to the involved component
+
+
+    """
+
+    if isinstance(logics, JsonLogicTest):
+        logic_dict = {
+            "nodeType": "operation",
+            "operator": logics.operator,
+            "children": [],
+        }
+        for index, value in enumerate(logics.values):
+
+            # There is an operation
+            if isinstance(value, JsonLogicTest):
+                if value.operator in ["var", "date"]:
+                    key = find_json_logic_test_value(logics.values[index])
+                    # leaf not related to component --> just a value compared
+                    if not component_map.get(key):
+                        logic_dict["value"] = key
+
+                    # Leaf is a field
+                    else:
+                        leaf_node = {
+                            "nodeType": "leaf",
+                            "key": key,
+                            "actualValue": resulting_data.get(key, ""),
+                            "stepName": component_map[
+                                key
+                            ].form_step.form_definition.name,
+                            "label": component_map[key].component.get("label", ""),
+                        }
+                        logic_dict["children"].append(leaf_node)
+                elif value.operator == "rdelta":
+                    logic_dict["value"] = value.values
+
+                # No relevant information with this operator
+                elif value.operator == "today":
+                    pass
+                else:
+                    logic_dict["children"].append(
+                        convert_json_logic_in_json_tree(
+                            value,
+                            component_map,
+                            resulting_data,
+                        )
+                    )
+
+            # No operation, it's a basic value
+            else:
+                logic_dict["value"] = value
+
+        return logic_dict
+
+    # array of rules
+    else:
+        return [
+            convert_json_logic_in_json_tree(logic, component_map, resulting_data)
+            for logic in logics
+        ]
```

Comment:
I'd swap the if statement to make use of a guard clause - treat the case where it isn't a `JsonLogicTest` first and get it out of the way, and then you can remove an indentation level (flat is better than nested!). So the structure becomes:

```py
if not isinstance(logics, JsonLogicTest):
    return [...]

logic_dict = {
    "nodeType": "operation",
    "operator": logics.operator,
    "children": [],
}
for index, value in enumerate(logics.values):
    ...
return logic_dict
```